### PR TITLE
Upgrade to Pro - Part 2 - Paddle integration

### PR DESCRIPTION
**Pull request by siddharthkp**
Part 2 of Upgrade integration

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/5357/commits/1b45f12d9ed0323fc5bdbcc6ef09bb1096bf873c is at https://pr5357.build.csb.dev/s/new.

File: packages/components/src/components/Menu/reach-menu.fork.tsx [hunk 1]
```diff
@@ -98,14 +98,22 @@ const initialState: MenuButtonState = {
...
   let menuRef = useRef(null);
   let popoverRef = useRef(null);
   let [descendants, setDescendants] = useDescendantsInit<
     MenuButtonDescendant
   >();
-  let [state, dispatch] = useReducer(reducer, initialState);
+
+  let [state, dispatch] = useReducer(reducer, {
+    ...initialState,
+    isExpanded: defaultOpen,
```
Comment (siddharthkp) [inline packages/components/src/components/Menu/reach-menu.fork.tsx:115]: new feature for default open :)

File: packages/app/src/app/pages/Pro/index-v2.tsx [hunk 2]
```diff
@@ -23,14 +23,16 @@ export const ProPage: React.FC = () => (
   <ThemeProvider>
     <Helmet>
       <title>Pro - CodeSandbox</title>
+      <script src="https://cdn.paddle.com/paddle/paddle.js" />
```
Comment (siddharthkp) [inline packages/app/src/app/pages/Pro/index-v2.tsx:26]: script tag, i know :(

File: packages/app/src/app/pages/Pro/index-v2.tsx [hunk 3]
```diff
@@ -39,44 +41,168 @@ export const ProPage: React.FC = () => (
...
+  name: string;
+  type: 'personal' | 'team';
+  frequency: 'monthly' | 'annual';
+  unit: number;
+  multiplier: number;
+  currency: string;
+};
+
+const PADDLE_VENDOR_ID = 729;
+const plans: { [key: string]: Plan } = {
```
Comment (siddharthkp) [inline packages/app/src/app/pages/Pro/index-v2.tsx:72]: this _config_ is still a work in progress, will move it to constants when it's shape is frozen

File: packages/app/src/app/pages/Pro/index-v2.tsx [hunk 4]
```diff
@@ -101,242 +227,522 @@ const Upgrade = () => {
...
 
-  const isPersonalPro = isPersonalWorkspace || user.subscription;
-  const isTeamPro = activeWorkspace.joinedPilotAt;
-
-  let currentPlanName: string;
-  if (isTeamPro) currentPlanName = 'Team Pro';
-  else if (isPersonalWorkspace && isPersonalPro) {
-    currentPlanName = 'Personal Pro';
-  } else currentPlanName = 'Community workspace (Free)';
+  const isLegacyPersonalPro = isPersonalWorkspace && user.subscription;
```
Comment (siddharthkp) [inline packages/app/src/app/pages/Pro/index-v2.tsx:235]: Existing Stripe users will become Legacy Pro. New pro users would fall in `workspace.subscription`

File: packages/app/src/app/pages/Pro/index-v2.tsx (hunk 2 above)
Comment (SaraVieira) [reply packages/app/src/app/pages/Pro/index-v2.tsx:26]: Then do something like this: https://github.com/codesandbox/codesandbox-client/blob/homepage-pro/packages/app/src/app/components/SubscribeForm/index.tsx#L31

File: packages/app/src/app/pages/Pro/index-v2.tsx [hunk 5]
```diff
@@ -39,44 +41,166 @@ export const ProPage: React.FC = () => (
...
-  }, [workspaceId, activeTeam, history, location, setActiveTeam]);
+  }, [workspaceId, activeTeam, history, location, setActiveTeam, type]);
+
+  const isPersonalWorkspace = personalWorkspaceId === activeTeam;
+  const [billingFrequency, setBillingFrequency] = React.useState<
+    Plan['frequency']
+  >('monthly');
+
+  React.useEffect(() => {
+    let newPlan: typeof plans[keyof typeof plans];
```
Comment (SaraVieira) [inline packages/app/src/app/pages/Pro/index-v2.tsx:192]: fancy TS man

File: packages/app/src/app/pages/Pro/index-v2.tsx [hunk 6]
```diff
@@ -95,248 +219,562 @@ const Upgrade = () => {
...
+      fontSize: 3,
+      width: 560,
+      marginTop: 120,
+      marginX: 'auto',
+      textAlign: 'center',
+    })}
+  >
+    <Icon name="simpleCheck" color="#5DCC67" size={64} />
+    <Text as="h1" size={8}>
+      You Payment is Successful
```
Comment (SaraVieira) [inline packages/app/src/app/pages/Pro/index-v2.tsx:765]: *was

File: packages/components/src/components/Menu/reach-menu.fork.tsx (hunk 1 above)
Comment (SaraVieira) [reply packages/components/src/components/Menu/reach-menu.fork.tsx:115]: 🔥

File: packages/app/src/app/pages/Pro/index-v2.tsx (hunk 2 above)
Comment (siddharthkp) [reply packages/app/src/app/pages/Pro/index-v2.tsx:26]: love it!

**Review by SaraVieira — APPROVED** [review]

**Comment by SaraVieira** [comment]
Ship it!!!! 💯

File: packages/app/src/app/pages/Pro/index-v2.tsx (hunk 5 above)
Comment (siddharthkp) [reply packages/app/src/app/pages/Pro/index-v2.tsx:194]: image

**Pull request merged**